Add screen orientation button

kazemihabib opened the pull request.

kazemihabib committed:
Add screen orientation button

GeoffreyMetais commented on vlc-android/src/org/videolan/vlc/gui/video/VideoPlayerActivity.java:
```diff
@@ -339,10 +341,10 @@ protected void onCreate(Bundle savedInstanceState) {
+        mOrientationToggle = findViewById(R.id.orientation_toggle);
 
         mScreenOrientation = Integer.valueOf(
-                mSettings.getString("screen_orientation", "99" /*SCREEN ORIENTATION SENSOR*/));
```
I strongly prefer to keep the sensors orientation by default.

GeoffreyMetais commented on vlc-android/src/org/videolan/vlc/gui/video/VideoPlayerActivity.java:
```diff
@@ -2493,6 +2500,9 @@ public void onChanged(@Nullable List<RendererItem> rendererItems) {
                 mHudBinding.playerOverlaySize.setVisibility(View.GONE);
             }
+
+            if (mScreenOrientation == 98)
```
maybe it could be visible in other cases, to trigger the manual mode

GeoffreyMetais commented:
Also, this button and options shouldn't be visible on TV!

kazemihabib committed:
Improve rotate button logic

Remove it from preferences as an orientation option
and give the orientation handling to rotate button
when that button is clicked and doesn't matter
what was selected then and also the button is not
visible on TV now

kazemihabib committed:
Add reset orientation and showInfo

GeoffreyMetais commented on vlc-android/res/values/strings.xml:
```diff
@@ -154,6 +154,9 @@
     <string name="audio_boost_warning">Slide up again to go beyond 100%</string>
+    <string name="locked_in_portrait_mode">Locked in portrait mode</string>
+    <string name="locked_in_landscape_mode">Locked in landscape mode</string>
+    <string name="reset_orientation">Reset Orientation</string>
```
"Orientation reset"

GeoffreyMetais commented on vlc-android/res/values/strings.xml:
```diff
@@ -154,6 +154,9 @@
     <string name="navmenu">navigation menu</string>
     <string name="advanced">advanced options</string>
     <string name="audio_boost_warning">Slide up again to go beyond 100%</string>
+    <string name="locked_in_portrait_mode">Locked in portrait mode</string>
```
Maybe we should add "(long click to reset to previous mode)"

Because feature would be hidden otherwise, but it makes the text a bit too long.

kazemihabib replied:
Maybe we should add sizes for void showInfo(String text, int duration) 
Like, small, medium, large (or pass directly the float size like the duration parameter)
and use small for longer messages.

kazemihabib committed:
Update strings and showInfo

GeoffreyMetais commented:
Merged
